### ISSUE TITLE
Using fcmpush behind a proxy

### ISSUE THREAD

**Issue by magynhard** [miyataka/fcmpush#38]
I i use fcmpush behind a proxy, the `http_proxy` or `https_proxy` environment variable is not considered, so the push will run into timeout:

```
A network error occurred: Net::OpenTimeout (Failed to open TCP connection to fcm.googleapis.com:443 (execution expired))
```

It would be nice, if i can set a proxy server in the fcmpush configuration, which is used for the requests, like this:

```ruby
Fcmpush.configure do |config|
  config.json_key_io = "#{Rails.root}/path/to/service_account_credentials.json"
  config.server_key = 'your firebase server key'
  config.proxy_server = 'http://myproxyserver:3128'
end
```

Otherwise if the environment variables `http_proxy` / `https_proxy` / `HTTP_PROXY` / `HTTPS_PROXY` would be used if set, might be another approach.

**Comment by miyataka** [comment]
@magynhard 
Thank you for your issue. I'll take a look later.
Also, Welcome to your contribution by pull-request.

Anyway, Thanks.

**Comment by CerfBarbu** [comment]
Hello, 

Can we have a TAG and new version of gem for this merge ? 

Thanks a lot

**Comment by miyataka** [comment]
@CerfBarbu 
FYI I've just released v1.4.2.
https://github.com/miyataka/fcmpush/releases/tag/v1.4.2
